src/main/java/org/cascadebot/cascadebot/commands/music/SkipCommand.java

```diff
@@ -0,0 +1,125 @@
+import org.cascadebot.cascadebot.commandmeta.Module;
+import org.cascadebot.cascadebot.permissions.CascadePermission;
+import org.cascadebot.cascadebot.utils.buttons.Button;
+import org.cascadebot.cascadebot.utils.votes.VoteButtonGroup;
+import org.cascadebot.cascadebot.utils.votes.VoteButtonGroupBuilder;
+import org.cascadebot.cascadebot.utils.votes.VoteMessageType;
+import org.cascadebot.cascadebot.utils.votes.VoteResult;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class SkipCommand implements ICommandMain {
+
+    private static Map<Long, VoteButtonGroup> voteMap = new HashMap<>();
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        AudioTrack track = context.getMusicPlayer().getPlayer().getPlayingTrack();
+        if(track == null) {
+            context.getTypedMessaging().replyDanger("I'm not playing anything!");
+            return;
+        }
+
+        if(context.getMusicPlayer().getTracks().peek() == null) {
+            context.getTypedMessaging().replyDanger("No song to skip to!");
+            return;
+        }
+
+        if(context.getArgs().length > 0) {
+            if (context.getArg(0).equalsIgnoreCase("force")) {
+                if(context.hasPermission("skip.force")) {
+                    context.getMusicPlayer().skip();
+                    context.getTypedMessaging().replySuccess("Forcefully skipped the song");
+                } else {
+                    context.getUIMessaging().sendPermissionError("skip.force");
+                }
+                return;
+            }
+        }
+
+        if(voteMap.containsKey(context.getGuild().getIdLong())) {
+            if(context.getArgs().length > 0) {
+                if(context.getArg(0).equalsIgnoreCase("yes")) {
+                    voteMap.get(context.getGuild().getIdLong()).addVote(sender.getUser(), UnicodeConstants.TICK);
+                    return;
+                } else if(context.getArg(0).equalsIgnoreCase("no")) {
+                    voteMap.get(context.getGuild().getIdLong()).addVote(sender.getUser(), UnicodeConstants.RED_CROSS);
+                    return;
+                }
+            }
+            voteMap.get(context.getGuild().getIdLong()).addVote(sender.getUser(), UnicodeConstants.TICK);
+            context.getTypedMessaging().replyWarning("A skip vote is already running, but we added your vote automatically");
+            return;
+        }
+
+        if(track.getUserData().equals(sender.getUser().getIdLong())) {
+            context.getMusicPlayer().skip();
+            context.getTypedMessaging().replySuccess("Skipped currently playing song because you queued it");
+            return;
+        }
+
+        VoteButtonGroupBuilder buttonGroupBuilder = new VoteButtonGroupBuilder(VoteMessageType.YES_NO);
+        buttonGroupBuilder.addExtraButton(new Button.UnicodeButton(UnicodeConstants.FAST_FORWARD, (runner, channel, message) -> {
+            //TODO check perm
```

Comment:
You have the context to check the permission with?

Reply:
that only check the cascade perm from the sender not anyone else afaik

Reply:
You can go ahead and add an override with a `member` param 👌

Reply:
lets just fix that in the permissions branch and I'll change it when we get things merged.

Reply:
Okay, could you still create a force skip perm?

Reply:
doing that right now